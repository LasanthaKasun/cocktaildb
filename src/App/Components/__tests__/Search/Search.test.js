import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Search } from "../..";

it("should render button component", () => {
  const component = renderer.create(
    <Search
      title="Search"
      icon={null}
      placeholder="Search"
      value=""
      onChangeText={null}
      onSearchAction={null}
    />
  );
  expect(component).toMatchSnapshot();
});

it("should render button with props", () => {
  render(
    <Search
      title="Search"
      icon={null}
      placeholder="Search"
      value=""
      onChangeText={null}
      onSearchAction={null}
    />
  );
  const search = screen.getByTestId("search");
  expect(search).toHaveTextContent("Search");
});
