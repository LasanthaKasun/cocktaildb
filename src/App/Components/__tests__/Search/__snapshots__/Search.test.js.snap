// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button component 1`] = `
<div
  className="w-11/12"
  data-testid="search"
>
  <div
    className="relative"
  >
    <div
      className="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"
    >
      <img
        alt="search"
        src={null}
        width="25"
      />
    </div>
    <input
      className="block w-full p-4 pl-12 text-sm text-gray-900 border border-gray-300 rounded-lg bg-gray-50 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
      defaultValue=""
      id="search"
      onChange={null}
      placeholder="Search"
      type="search"
    />
    <button
      className="text-white absolute right-2.5 bottom-2.5 bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-4 py-2 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
      onClick={null}
      type="submit"
    >
      Search
    </button>
  </div>
</div>
`;
